model/Aktivita/Aktivita.php

```diff
@@ -1945,12 +1946,16 @@ public function zkontrolujZdaSeMuzePrihlasit(
             }
         }
 
-        // přihlášení na navázané aktivity (jen pokud není teamleader)
-        if ($this->a['dite'] && $this->pocetPrihlasenych() > 0) {
+        // přihlášení na navázané aktivity
+        if ($this->a['dite']) {
             $deti = $this->deti();
             if (count($deti) === 1) {
-                current($deti)->prihlas($uzivatel, $prihlasujici, self::STAV | ($parametry & self::UKAZAT_DETAILY_CHYBY));
-            } else {
+                try {
+                    current($deti)->prihlas($uzivatel, $prihlasujici, self::STAV | ($parametry & self::UKAZAT_DETAILY_CHYBY));
```

Comment:
Je to sice nepravděpodobná situace, ale v PHP může být `current` nebezpečný, protože někdo posunul pointer v array, například přes https://www.php.net/manual/en/function.next.php, a pak u pole s jedním prvkem vrátí `current -> false`

Použij `reset`
```sh
php -a 
Interactive shell

php > $a = [1];
php > next($a);
php > var_dump(current($a));
bool(false)
php > var_dump(reset($a));
int(1)
php > 
```